homework-submission/todoapp/src/Components/TodoElement.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { observer, inject } from 'mobx-react';
+
+@inject('todos')
+@observer
+class TodoItem extends React.Component {
+  state = {
+    edit: false
+  }
+  toggleHandler = () => {
+    this.setState({
+      edit: !this.state.edit
+    })
+  }
+  handleUpdate = (e) => {
+    e.preventDefault()
+    const todo = this.props.todo
+    const { updateHandler } = this.props.todos
+    let newTask = this.editField.value
+    updateHandler(todo._id, newTask)
+  }
+  render() {
+    const todo = this.props.todo;
+    const { deleteHandler, checkBoxHandler } = this.props.todos;
+    let TodoView;
+    if (!this.state.edit) {
+      TodoView = (
+        <div className="todo">
+          <input type='checkbox' defaultChecked={todo.done} onChange={() => checkBoxHandler(todo._id)} />
```

Comment:
You don't have any bugs with this But it Would be as you saw before an improvement avoiding them, Please have another look here: https://github.com/HackYourFuture/React/pull/280/files#r236788358